discretisedfield/plotting/mpl_field.py

```diff
@@ -601,12 +590,16 @@ def vector(self,
 
         multiplier = self._setup_multiplier(multiplier)
 
-        points, values = map(list, zip(*list(self.field)))
+        points1 = self.field.mesh.midpoints[self.axis1] / multiplier
+        points2 = self.field.mesh.midpoints[self.axis2] / multiplier
 
-        # remove points and values where norm is 0.
-        points = [p for p, v in zip(points, values)
-                  if not np.equal(v, 0).all()]
-        values = [v for v in values if not np.equal(v, 0).all()]
+        values = self.field.array.copy().reshape(self.n + (self.field.dim,))
+        # filter out points where norm is 0
+        values[np.all(np.isclose(values, 0, atol=0), axis=-1)] = np.nan
```

Comment:
`rtol` and `atol` are actually added together internally, it is not either/or. Please check here: https://numpy.org/doc/stable/reference/generated/numpy.isclose.html.

Reply:
I just did. Do you think we need to change something here? I've basically kept the checks in this line as they were in the old implementation.

Reply:
That is fine as long you are aware there is still rtol there